installer/assets/install

```diff
@@ -0,0 +1,145 @@
+#!/bin/sh
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+set -eu
+
+usage() {
+  cat 1>&2 <<EOF
+Installs the AWS SAM CLI Linux executable package
+USAGE:
+    install [FLAGS] [OPTIONS]
+FLAGS:
+    -u, --update              Updates the AWS SAM CLI if a different version
+                              is previously installed. By default, this script
+                              will not update the AWS SAM CLI if a previous
+                              installation is detected.
+    -h, --help                Prints help information
+OPTIONS:
+    -i, --install-dir <path>  The directory to install the AWS SAM CLI. By
+                              default, this directory is: /usr/local/aws-sam-cli
+    -b, --bin-dir <path>      The directory to store symlinks to executables
+                              for the AWS SAM CLI. By default, the directory
+                              used is: /usr/local/bin
+EOF
+}
+
+parse_commandline() {
+  while test $# -gt 0
+  do
+    key="$1"
+	case "$key" in
+	  -i|--install-dir)
+	    PARSED_INSTALL_DIR="$2"
+		shift
+	   ;;
+	  -b|--bin-dir)
+	    PARSED_BIN_DIR="$2"
+		shift
+	   ;;
+	  -u|--update)
+	    PARSED_UPGRADE="yes"
+	  ;;
+	  -h|--help)
+	    usage
+        exit 0
+	  ;;
+	  *)
+	   die "Got an unexpected argument: $1"
+	  ;;
+    esac
+	shift
+  done
+}
+
+set_global_vars() {
+  ROOT_INSTALL_DIR=${PARSED_INSTALL_DIR:-/usr/local/aws-sam-cli}
+  BIN_DIR=${PARSED_BIN_DIR:-/usr/local/bin}
+  UPGRADE=${PARSED_UPGRADE:-no}
+
+  EXE_NAME="sam"
+  INSTALLER_DIR="$( cd "$( dirname "$0" )" >/dev/null 2>&1 && pwd )"
+  INSTALLER_DIST_DIR="$INSTALLER_DIR/dist"
+  INSTALLER_EXE="$INSTALLER_DIST_DIR/$EXE_NAME"
+  SAM_EXE_VERSION=$($INSTALLER_EXE --version | cut -d ' ' -f 4)
+
+  INSTALL_DIR="$ROOT_INSTALL_DIR/$SAM_EXE_VERSION"
+  INSTALL_DIR="$INSTALL_DIR"
+  INSTALL_DIST_DIR="$INSTALL_DIR/dist"
+  INSTALL_BIN_DIR="$INSTALL_DIR/bin"
+  INSTALL_SAM_EXE="$INSTALL_BIN_DIR/$EXE_NAME"
+
+  CURRENT_INSTALL_DIR="$ROOT_INSTALL_DIR/current"
+  CURRENT_SAM_EXE="$CURRENT_INSTALL_DIR/bin/$EXE_NAME"
+
+  BIN_SAM_EXE="$BIN_DIR/$EXE_NAME"
+}
+
+create_install_dir() {
+  mkdir -p "$INSTALL_DIR" || exit 1
```

Comment:
When is `mkdir -p` expected to fail? Even if it does `set -e` will automatically abort the script.